# Stringtie

Base directory to store Stringtie downloads scripts and config files

```
Stringtie:

StringTie is a fast and highly efficient assembler of RNA-Seq alignments into potential transcripts. 
It uses a novel network flow algorithm as well as an optional de novo assembly step to assemble and 
quantitate full-length transcripts representing multiple splice variants for each gene locus. 

Its input can include not only the alignments of raw reads used by other transcript assemblers, 
but also alignments longer sequences that have been assembled from those reads.

In order to identify differentially expressed genes between experiments,
StringTie's output can be processed by specialized software like Ballgown, 
Cuffdiff or other programs (DESeq2, edgeR, etc.).

The main input of the program is a BAM file with RNA-Seq read mappings 
which must be sorted by their genomic location (for example the accepted_hits.
bam file produced by TopHat or the output of HISAT2 after sorting and converting it using samtools). 

```
## Usage

See: http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual#run

## Upgrades Frequency

See:

## How to Install Stringtie
See: http://ccb.jhu.edu/software/stringtie/#install
  
